src/components/advanced-search/advanced-search-form.vue

```diff
@@ -158,6 +160,15 @@ export default class AdvancedSearchForm extends Vue {
     if (!this.applicantFirstName) this.applicantLastNameRules = []
   }
 
+  /** Clear form. */
+  @Watch('formReset')
+  private clearForm (): void {
```

Comment:
Would you want to do this only when `formReset` becomes True? (ie, `if (val) then ...`)

Reply:
Not exactly. 
In this case i want to do this when the dialog closes or opens, so when the dialog state changes.

Reply:
Is `formReset` tied to the dialog opening or closing?

Reply:
Oh, it is...
```
:formReset="dialog"
```

Reply:
Then why not call the prop something more topical?

As I reads now, I would expect this property to reset the form only when it's set to True.

Reply:
Sure

Reply:
A name that makes more sense to me is "resetOnToggle" (but I don't like this name either 😛 ).

Reply:
Updated 👍

Reply:
Where I'm coming from is this...

From the parent's point of view, it's informing the component when `dialog` changes. The prop name should reflect that.

From the component's point of view, what it does with that prop (ie, reset the form) is an internal thing.

But a prop named "formReset" implies that the form should be reset when it becomes True.

Reply:
For sure! Updated to be more topical.